nomad-base/src/merkle_sync.rs

```diff
@@ -0,0 +1,101 @@
+use color_eyre::Result;
+use ethers::types::H256;
+use nomad_core::{
+    accumulator::incremental::IncrementalMerkle, db::DbError, ChainCommunicationError,
+};
+use std::{sync::Arc, time::Duration};
+use tokio::{sync::RwLock, task::JoinHandle, time::sleep};
+use tracing::{debug, error, info, info_span, instrument::Instrumented, Instrument};
+
+use crate::NomadDB;
+
+/// Self-syncing light merkle tree. Polls for new messages and updates tree.
+#[derive(Debug, Clone)]
+pub struct IncrementalMerkleSync {
```

Comment:
Will change to `LightMerkleSync` and use "light" in favor of "incremental" after #35 merged